moto/applicationautoscaling/models.py

```diff
@@ -0,0 +1,223 @@
+from __future__ import unicode_literals
+from moto.core import BaseBackend, BaseModel
+from moto.ecs import ecs_backends
+from .exceptions import AWSValidationException
+from collections import OrderedDict
+from enum import Enum, unique
+import time
+
+
+@unique
+class ServiceNamespaceValueSet(Enum):
+    APPSTREAM = "appstream"
+    RDS = "rds"
+    LAMBDA = "lambda"
+    CASSANDRA = "cassandra"
+    DYNAMODB = "dynamodb"
+    CUSTOM_RESOURCE = "custom-resource"
+    ELASTICMAPREDUCE = "elasticmapreduce"
+    EC2 = "ec2"
+    COMPREHEND = "comprehend"
+    ECS = "ecs"
+    SAGEMAKER = "sagemaker"
+
+
+@unique
+class ScalableDimensionValueSet(Enum):
+    CASSANDRA_TABLE_READ_CAPACITY_UNITS = "cassandra:table:ReadCapacityUnits"
+    CASSANDRA_TABLE_WRITE_CAPACITY_UNITS = "cassandra:table:WriteCapacityUnits"
+    DYNAMODB_INDEX_READ_CAPACITY_UNITS = "dynamodb:index:ReadCapacityUnits"
+    DYNAMODB_INDEX_WRITE_CAPACITY_UNITS = "dynamodb:index:WriteCapacityUnits"
+    DYNAMODB_TABLE_READ_CAPACITY_UNITS = "dynamodb:table:ReadCapacityUnits"
+    DYNAMODB_TABLE_WRITE_CAPACITY_UNITS = "dynamodb:table:WriteCapacityUnits"
+    RDS_CLUSTER_READ_REPLICA_COUNT = "rds:cluster:ReadReplicaCount"
+    RDS_CLUSTER_CAPACITY = "rds:cluster:Capacity"
+    COMPREHEND_DOCUMENT_CLASSIFIER_ENDPOINT_DESIRED_INFERENCE_UNITS = (
+        "comprehend:document-classifier-endpoint:DesiredInferenceUnits"
+    )
+    ELASTICMAPREDUCE_INSTANCE_FLEET_ON_DEMAND_CAPACITY = (
+        "elasticmapreduce:instancefleet:OnDemandCapacity"
+    )
+    ELASTICMAPREDUCE_INSTANCE_FLEET_SPOT_CAPACITY = (
+        "elasticmapreduce:instancefleet:SpotCapacity"
+    )
+    ELASTICMAPREDUCE_INSTANCE_GROUP_INSTANCE_COUNT = (
+        "elasticmapreduce:instancegroup:InstanceCount"
+    )
+    LAMBDA_FUNCTION_PROVISIONED_CONCURRENCY = "lambda:function:ProvisionedConcurrency"
+    APPSTREAM_FLEET_DESIRED_CAPACITY = "appstream:fleet:DesiredCapacity"
+    CUSTOM_RESOURCE_RESOURCE_TYPE_PROPERTY = "custom-resource:ResourceType:Property"
+    SAGEMAKER_VARIANT_DESIRED_INSTANCE_COUNT = "sagemaker:variant:DesiredInstanceCount"
+    EC2_SPOT_FLEET_REQUEST_TARGET_CAPACITY = "ec2:spot-fleet-request:TargetCapacity"
+    ECS_SERVICE_DESIRED_COUNT = "ecs:service:DesiredCount"
+
+
+class ApplicationAutoscalingBackend(BaseBackend):
+    def __init__(self, region, ecs):
+        super(ApplicationAutoscalingBackend, self).__init__()
+        self.region = region
+        self.ecs_backend = ecs
+        self.targets = OrderedDict()
+
+    def reset(self):
+        region = self.region
+        ecs = self.ecs_backend
+        self.__dict__ = {}
+        self.__init__(region, ecs)
+
+    @property
+    def applicationautoscaling_backend(self):
+        return applicationautoscaling_backends[self.region]
+
+    def delete_scaling_policy(self):
```

Comment:
Please remove any methods not yet implemented. Moto will raise a very helpful 'NotYetImplemented' exception - as long as the method does not exist. 
Having the method but not giving any meaningful response will be confusing. Same goes for these methods in response.py

Reply:
Thanks @bblommers, will do.